.github/workflows/ci.yml

```diff
@@ -163,3 +181,42 @@ jobs:
         run: |
           make gomobileinit
           (cd $GOPATH/$GO_SRC_DIR; make ios)
+
+  report-artifacts:
+    needs: [android, qt-linux, macos]
+    runs-on: ubuntu-22.04
+    if: ${{ !cancelled() && github.event_name == 'push' }}
+    steps:
+      - name: Clone the repo
+        uses: actions/checkout@v4
+      - name: Create vars
+        id: vars
+        run: |
+          echo "git_sha_short=$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
+      - name: Message for success
+        uses: mattermost/action-mattermost-notify@master
+        continue-on-error: true
+        if: job.status == 'success'
+        with:
+          MATTERMOST_WEBHOOK_URL: ${{ secrets.MM_WEBHOOK_URL }}
+          PAYLOAD: |-
+            {
+              "channel": "eng-artifacts",
+              "text": "**New artifacts built**\n([${{ github.ref_name }}](https://github.com/${{ github.repository }}/tree/${{ github.ref_name }}), [${{ steps.vars.outputs.git_sha_short }}](https://github.com/${{ github.repository }}/commit/${{ github.sha }}), [artifacts](https://github.com/${{ github.repository }}/actions/runs/${{github.run_id}}#artifacts))\n* Android - [APK](${{needs.android.outputs.artifact-url}})\n* Linux - [AppImage](${{needs.qt-linux.outputs.artifact-url-ai}}), [DEB](${{needs.qt-linux.outputs.artifact-url-deb}}), [RPM](${{needs.qt-linux.outputs.artifact-url-rpm}})\n* MacOS - [Zip](${{needs.macos.outputs.artifact-url}})",
+              "icon_emoji": "white_check_mark",
+              "props": {"card": "Repository: [${{ github.repository }}](https://github.com/${{ github.repository }})"}
+            }
+
+      - name: Message for failure
+        uses: mattermost/action-mattermost-notify@master
+        continue-on-error: true
+        if: job.status == 'failure'
+        with:
+          MATTERMOST_WEBHOOK_URL: ${{ secrets.MM_WEBHOOK_URL }}
+          PAYLOAD: |-
+            {
+              "channel": "eng-build-failures",
```

Comment:
does anyone need this? people working on their PRs will check their PRs.

Reply:
This is in case the main branch fails to build. Which hopefully never happens.

Reply:
oh wait this is for pushes to branches in this repo too, nvm